encoders/firebase-decoders-json/src/test/java/com/google/firebase/decoders/TypeTokenTest.java

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.decoders;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(JUnit4.class)
+public class TypeTokenTest {
+  static class Foo {}
+
+  //Primitive type
+  @Test
+  public void primitiveType_typeIsCorrectlyCaptured(){
+    ClassToken<?> intToken = (ClassToken<?>) TypeToken.of(int.class);
```

Comment:
Instead of unconditionally casting to `ClassToken` I suggest you make explicit checks for instanceof, this will make test failures more clear and explicit.
I'd recommend a pattern like:

```
TypeToken<Integer> token = TypeToken.of(int.class);
assertThat(token).isInstanceOf(ClassToken.class);

ClassToken<Integer> intToken = (ClassToken<Integer>) token;
// test the rest
```

Additionally it might not be required to test all primitives, a couple should be enough.